src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -2128,6 +2128,12 @@ public FixableDiagnostic ProviderDeclarationViaImportKeywordIsDeprecated(Provide
                     DiagnosticStyling.Default,
                     codeFix);
             }
+
+            public ErrorDiagnostic ArtifactFilePathCouldNotBeResolved(string ociManifestPath) => new(
+                TextSpan,
+                "BCP382",
+                $"The artifact file path could not be resolved from the OCI manifest file: \"{ociManifestPath}\"."
```

Comment:
I don't think this error is going to be understandable for an end-user - I expect many type users won't know what "OCI", "manifest" or "artifact" means.

I'd suggest starting with something more generic like "Provider package {foo} is malformed and could not be loaded."